rs/sns/init/src/lib.rs

```diff
@@ -3471,61 +3444,25 @@ mod test {
         }
         // Test that if the constraints are violated, the validation, indeed, fails.
         {
-            let fdvp = FractionalDVP {
-                swap_distribution: Some(SwapDistribution {
-                    initial_swap_amount_e8s: MAX_DIRECT_ICP_CONTRIBUTION_TO_SWAP,
-                    // Not used in this test.
-                    total_e8s: 0,
-                }),
-                // Not used in this test.
-                developer_distribution: None,
-                treasury_distribution: None,
-                airdrop_distribution: None,
-            };
             let sns_init_payload = SnsInitPayload {
-                max_direct_participation_icp_e8s: Some(MAX_DIRECT_ICP_CONTRIBUTION_TO_SWAP),
-                min_direct_participation_icp_e8s: Some(1),
                 // This value being so low ensures there would be too many SNS neurons.
-                min_participant_icp_e8s: Some(2),
-                max_participant_icp_e8s: Some(MAX_DIRECT_ICP_CONTRIBUTION_TO_SWAP),
-                min_participants: Some(40_000),
-                initial_token_distribution: Some(FractionalDeveloperVotingPower(fdvp)),
-                neuron_basket_construction_parameters: Some(NeuronBasketConstructionParameters {
-                    count: 2,
-                    // Not used in this test.
-                    dissolve_delay_interval_seconds: 0,
-                }),
-                neuron_minimum_stake_e8s: Some(1),
-                transaction_fee_e8s: Some(0),
-                ..SnsInitPayload::with_valid_values_for_testing_pre_execution()
+                min_participant_icp_e8s: Some(1),
+                ..sns_init_payload
             };
 
             // Run code under test.
             let error = sns_init_payload
                 .validate_participation_constraints()
                 .unwrap_err();
             {
-                let expected_error_fragment =
-                    "number of SNS neurons created for direct participants";
+                let expected_error_fragment = "min_participant_icp_e8s=1 is too small.";
```

Comment:
Does the error also say why it's too small? I suspect the reason is not related to number of neurons.

Reply:
> I suspect the reason is not related to number of neurons.

Not anymore.

Are you suggesting to make the expected text more verbose?

Reply:
made the assertion more precise